packages/@react-native-windows/codegen/src/generators/PropObjectTypes.ts

```diff
@@ -0,0 +1,233 @@
+import type { CppCodegenOptions } from './ObjectTypes';
+import {
+  AliasMap,
+  getAnonymousAliasCppName,
+} from './AliasManaging';
+
+
+// eslint-disable-next-line complexity
+export function translateComponentPropsFieldType(type: PropTypeAnnotation,
+  aliases: AliasMap<ObjectTypeAnnotation<PropTypeAnnotation>>,
+  baseAliasName: string,
+  options: CppCodegenOptions): { type: string, initializer: string, alreadySupportsOptionalOrHasDefault?: boolean } {
+  switch (type.type) {
+    case 'StringTypeAnnotation':
+      return { type: options.cppStringType, initializer: type.default ? `{${type.default}}` : '', alreadySupportsOptionalOrHasDefault: !!type.default };
+    case 'FloatTypeAnnotation':
+      return { type: 'float', initializer: type.default ? `{${type.default}}` : '{}', alreadySupportsOptionalOrHasDefault: !!type.default };
+    case 'DoubleTypeAnnotation':
+      return { type: 'double', initializer: type.default ? `{${type.default}}` : '{}', alreadySupportsOptionalOrHasDefault: !!type.default };
+    case 'Int32TypeAnnotation':
+      return { type: 'int32_t', initializer: type.default ? `{${type.default}}` : '{}', alreadySupportsOptionalOrHasDefault: !!type.default };
+    case 'BooleanTypeAnnotation':
+      return { type: 'bool', initializer: type.default ? `{${type.default}}` : '{}', alreadySupportsOptionalOrHasDefault: !!type.default };
+    case 'ArrayTypeAnnotation':
+
+      let arrayTemplateArg = '';
+      switch (type.elementType.type) {
+        case 'BooleanTypeAnnotation':
+          arrayTemplateArg = 'bool';
+          break;
+        case 'DoubleTypeAnnotation':
+          arrayTemplateArg = 'double';
+          break;
+        case 'FloatTypeAnnotation':
+          arrayTemplateArg = 'float';
+          break;
+        case 'Int32TypeAnnotation':
+          arrayTemplateArg = 'int32_t';
+          break;
+        case 'StringTypeAnnotation':
+          arrayTemplateArg = options.cppStringType;
+          break;
+        case 'ArrayTypeAnnotation':
+          const innerType = translateComponentPropsFieldType(type.elementType, aliases, baseAliasName, options);
+          arrayTemplateArg = `std::vector<${innerType.type}>`;
+          break;
+        case 'ObjectTypeAnnotation':
+          arrayTemplateArg = translateComponentPropsFieldType(type.elementType, aliases, baseAliasName, options).type;
+          break;
+        case 'ReservedPropTypeAnnotation':
+          switch (type.elementType.name) {
+            case 'ColorPrimitive':
+              arrayTemplateArg = 'winrt::Microsoft::ReactNative::Color';
+              break;
+            case 'DimensionPrimitive':
+            case 'EdgeInsetsPrimitive':
+            case 'ImageRequestPrimitive':
+            case 'ImageSourcePrimitive':
+            case 'PointPrimitive':
+              arrayTemplateArg = 'winrt::Microsoft::ReactNative::JSValue'; // TODO - better handling for these types than JSValue
```

Comment:
do we need tracking issues for these TODOs?

Reply:
Filed: https://github.com/microsoft/react-native-windows/issues/13658